crates/dojo-world/src/migration/world.rs

```diff
@@ -107,6 +113,63 @@ impl WorldDiff {
         count += self.contracts.iter().filter(|s| !s.is_same()).count();
         count
     }
+
+    pub fn update_order(&mut self) -> Result<()> {
+        let mut ts = TopologicalSort::<&str>::new();
+
+        // make the dependency graph by reading the constructor_calldata
+        for contract in self.contracts.iter() {
+            let curr_name: &str = &contract.name;
+            ts.insert(curr_name);
+
+            for field in &contract.constructor_calldata {
+                if let Some(dependency) = field.strip_prefix("$contract_address:") {
+                    ts.add_dependency(dependency, curr_name);
+                } else if let Some(dependency) = field.strip_prefix("$class_hash:") {
+                    ts.add_dependency(dependency, curr_name);
+                } else {
+                    // verify its a field element
+                    match FieldElement::from_str(field) {
+                        Ok(_) => continue,
+                        Err(e) => bail!(format!(
+                            "Expected constructor_calldata element to be a special variable (i.e. \
+                             starting with $contract_address or $class_hash) or be a \
+                             FieldElement.Failed with error: {e:?}"
```

Comment:
```suggestion
                             FieldElement. Failed with error: {e:?}"
```
We may have the verbose here to print the debug error?